arch/arm/core/cortex_m/isr_wrapper.S

```diff
@@ -97,13 +97,24 @@ _idle_state_cleared:
 
 #endif /* CONFIG_PM */
 
+#if defined(CONFIG_ARM_CUSTOM_INTERRUPT_CONTROLLER)
+	bl z_soc_irq_get_active
+#else
 	mrs r0, IPSR	/* get exception number */
+#endif /* CONFIG_ARM_CUSTOM_INTERRUPT_CONTROLLER */
+
 #if defined(CONFIG_ARMV6_M_ARMV8_M_BASELINE)
 	ldr r1, =16
 	subs r0, r1	/* get IRQ number */
+#if defined(CONFIG_ARM_CUSTOM_INTERRUPT_CONTROLLER)
```

Comment:
Do you think we can use other registers or any other method to avoid this guard?

Reply:
Would'nt it be simpler and make more sense to completely opt-out of this file if a custom irq controller is used and let the custom implementation handle routing to the isr handler?

Reply:
> Would'nt it be simpler and make more sense to completely opt-out of this file if a custom irq controller is used and let the custom implementation handle routing to the isr handler?

I think there is other functionality in this file that we probably wouldn't want to force users to exclude or re-implement. For example, the tracing and PM handling.

Reply:
> Do you think we can use other registers or any other method to avoid this guard? Just feel like there's so many guards.

Yeah, maybe I can stash this in a calle save register, Ill look into it.

Reply:
I updated the PR to stach the number in r4, it gets rid of one of the guards and eliminates the use of stack.